example/exampleContour.html

```diff
@@ -1,123 +1,127 @@
+      
+      const ContourChart = dc.rasterChart(parent, true, null, mapboxgl)
       .con(con)
       .height(h)
       .width(w)
       .mapUpdateInterval(750)
       .mapStyle('mapbox://styles/mapbox/light-v8')
       .mapboxToken(mapboxToken) // need a mapbox accessToken for loading the tiles
       .popupSearchRadius(2)
-      .useLonLat(true)
       .useGeoTypes(true) // need for projecting geo column using "mercator_map_projection"
-
+      
+      const isMajorFieldName = "is_major"
+      
       const contourPolygonLayer = dc.rasterLayer("polys")
       .crossfilter(crossFilter)
       .setState({
-          data: [
-            {
-              type: "contour",
-              table: config.table,
-              source: config.table,
-              contour_value_field: config.value,
-              lat_field: config.lat_field,
-              lon_field: config.lon_field,
-              agg_type: 'AVG',
-              fill_agg_type: 'AVG',
-              bin_dim_meters: config.bin_dim_meters,
-              neighborhood_fill_radius: 0,
-              fill_only_nulls: false,
-              flip_latitude: false,
-              minor_contour_interval: config.minor_contour_interval,
-              major_contour_interval: config.major_contour_interval,
-              contour_offset: 0,
-            }
-          ],
-          transform: {
-            projection: "mercator_map_projection",
-          },
-          enableHitTesting: false,
-          mark: {
-            type: "poly",
-            strokeColor: "white",
-            strokeWidth: 0,
-            fillColor: "",
-            lineJoin: "miter",
-            miterLimit: 10
-          },
-          encoding: {
-            color: {
-              type: "linear",
-              domain: [0, 6000],
-              range: ["#FF0000", "#0000FF"],
-              opacity: 0.50
-            }
+        data: [
+        {
+          type: "contour",
+          table: config.table,
+          source: config.table,
+          contour_value_field: config.value,
+          lat_field: config.lat_field,
+          lon_field: config.lon_field,
+          agg_type: 'AVG',
+          fill_agg_type: 'AVG',
+          bin_dim_meters: config.bin_dim_meters,
+          neighborhood_fill_radius: 0,
+          fill_only_nulls: false,
+          flip_latitude: false,
+          contour_offset: 0,
+          intervals: {
+            isMajorFieldName,
+            major: config.major_contour_interval,
+            minor: config.minor_contour_interval
           }
-        })
-
+        }
+        ],
+        transform: {
+          projection: "mercator_map_projection",
+        },
+        mark: {
+          type: "poly",
+          strokeColor: "white",
+          strokeWidth: 0,
+          fillColor: "",
+          lineJoin: "miter",
+          miterLimit: 10
+        },
+        encoding: {
+          geocol: "contour_polygons",
+          color: {
```

Comment:
Yeah I believe it uses `geocol` if provided for this, not totally clear on the magic behind this though

Reply:
ok my answer made no sense, i think this question is going over my head a bit... may need to trace it through the polygon mixin a bit to find out

Reply:
yeah, seems like there's a hole here as well that will need filling. it can basically act the same as the line implementation tho. Again, I don't mind you landing as is and fixing in a follow-up.

How does this example color the polygons tho?

Reply:
AH, contours creates its own scales, and uses the `contour_value_field` as the field for polygons. It generates:
```
  "scales": [
    {
      "name": "contour_fill",
      "type": "linear",
      "domain": [0.0, 4000.0],
      "range": ["red", "blue"]
    }
  ],
...
"marks": [
    {
      "type": "polys",
      "from": {"data": "contourPolygonLayer"},
      "properties": {
        "x": {"field": "x"},
        "y": {"field": "y"},
        "fillColor": {"field": "contour_values", "scale": "contour_fill"},
        "strokeColor": "white",
        "strokeWidth": 0,
        "lineJoin": "bevel"
      },
      "transform": {"projection": "mercator_map_projection"}
    }
  ]
```

Reply:
Gonna merge once tests pass, and we can continue convos in follow ups, i have one or two things left for contour charting